### PR TITLE
Update CSV Result Code Validation 

### LINKED ISSUE #7898: Update Bulk Result File Validation: Result Code
File validation for the bulk result uploader was a contributing factor to the bug described in #7897.

Requested update:

Adjust the validation for the `test_result` field in the bulk uploader to be more stringent.

[/backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java#L273-L275](https://github.com/CDCgov/prime-simplereport/blob/268ddb50c86572fc83419243cc36a45dfd4845a1/backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java#L273-L275)
```java
  public static List<FeedbackMessage> validateTestResult(ValueOrError input) {
    return validateSpecificValueOrSNOMED(input, TEST_RESULT_VALUES);
  }
```

Rather than allowing anything that looks like a snomed (i.e. the regex currently in place), we should validate against a list of specific snomeds that we support and can effectively map in the fhir converter (which should include any snomeds in the [covid](https://prime.cdc.gov/api/metadata/livd) or [hiv](https://prime.cdc.gov/api/metadata/livd) livd tables).

The SNOMEDs we allow should be the same ones being explicitly mapped to normal/abnormal flags in #7897

### PR DESCRIPTION
**Pull request by bobbywells52**
# BACKEND PULL REQUEST

## Related Issue
resolves #7898 

## Changes Proposed

- Adjust the validation for the `test_result` field in the bulk uploader to only validate against a static list of SNOMEDs or string literals that map to a specific SNOMED.

## Additional Information

- @emyl3 pointed out in [the comments for this ticket](https://github.com/CDCgov/prime-simplereport/issues/7898#issuecomment-2223933627) that some of these SNOWMEDs while supported by SR will make the COVID pipeline fail as certain codes that are used as results for STD's are not valid codes on the COVID pipeline. That issue is not addressed in this PR and I will be creating a new ticket where that issue can be addressed. 

## Testing

- Can be tested on either a local version of this branch or on dev5 by uploading a CSV to the bulk uploader. 
- For the `test_results` row, all cells with any of [these SNOMEDs](https://github.com/CDCgov/prime-reportstream/blob/1ffae4ca0b04cd0aa9f169e26813ecd86df71bb5/prime-router/src/main/kotlin/metadata/Mappers.kt#L768) should pass along with any of the result SNOMEDs that we are using for HIV and Syphilis as well. Any other SNOMED should cause the upload to fail and alert the user of the value that is failing.

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java [hunk 1]
```diff
@@ -604,17 +609,23 @@ public static String convertSexToDatabaseValue(String biologicalSex) {
...
     List<FeedbackMessage> errors = new ArrayList<>();
     String value = parseString(input.getValue());
+
     if (value == null) {
       return errors;
     }
     boolean nonSNOMEDValue = value.matches(ALPHABET_REGEX);
+    boolean snomedValue = value.matches(SNOMED_REGEX);
+
     if (nonSNOMEDValue) {
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java:622]: This logic wouldn't error for some invalid test result values that have been used in prod recently e.g. (`Positive, 10828004` and `-`).
Screenshot 2024-07-24 at 10 13 34

**Review by emyl3 — CHANGES_REQUESTED** [review]
Left one comment about the validation logic.

File: backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java (hunk 1 above)
Comment (bobbywells52) [reply backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java:622]: @emyl3 From my understanding, the point of this ticket is to make it such that `10828004` and `Positive` are supported values. I did not consider non letter chars in the filtering though -- will update that now.

File: backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java [hunk 2]
```diff
@@ -182,9 +184,15 @@ public class CsvValidatorUtils {
           "y", "yes",
           "n", "no",
           "u", UNKNOWN_CODE);
-  private static final Set<String> TEST_RESULT_VALUES =
+  private static final Set<String> ACCEPTED_LITERAL_TEST_RESULT_VALUES =
       Set.of(POSITIVE_LITERAL, "negative", "not detected", DETECTED_LITERAL, "invalid result");
```
Comment (fzhao99) [inline backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java:188]: nit: could we replace these loose strings with variables like we have for the positive and detected cases? Would rather not have the inconsistencies here

**Review by fzhao99 — APPROVED** [review]

**Review by mpbrown — APPROVED** [review]
LGTM! Nice work on this Bobby!

**Review by emyl3 — APPROVED** [review]
Thanks for the additional tests and fix!

**Pull request merged**